### PR TITLE
feat: add java jvs integration

### PR DESCRIPTION
**Pull request by raserva**
integration tests to come.

### CODE REVIEW

File: clients/java-logger/library/src/main/java/com/abcxyz/lumberjack/auditlogclient/AuditLoggingServerInterceptor.java [hunk 1]
```diff
@@ -116,6 +122,22 @@ public <ReqT, RespT> Listener<ReqT> interceptCall(
...
       next.startCall(call, headers);
     }
+
+    if (auditLoggingConfiguration.isJustificationRequired()) {
+      try {
+        logBuilder.setMetadata(getStructForJustification(headers));
+      } catch (JwkException e) {
+        log.warn("Exception while trying to determine justification.");
+        if (ConfigUtils.shouldFailClose(auditLoggingConfiguration.getLogMode())) {
+          throw new IllegalStateException("Unable to determine justificationjustificationjustificationjustificationjustificationjustificationjustificationjustificationjustification.", e);
```
[the remark this answers is not in the document]
Comment (raserva) [reply clients/java-logger/library/src/main/java/com/abcxyz/lumberjack/auditlogclient/AuditLoggingServerInterceptor.java:132]: LOL no idea how this happened

[the remark this answers is not in the document]
Comment (raserva) [reply clients/java-logger/library/src/main/java/com/abcxyz/lumberjack/auditlogclient/AuditLoggingServerInterceptor.java]: i just really like the word justification

[the remark this answers is not in the document]
Comment (myurtoglu) [reply clients/java-logger/library/src/main/java/com/abcxyz/lumberjack/auditlogclient/AuditLoggingServerInterceptor.java]: Maybe you should take a break from working on JVS a bit :rofl:

File: clients/java-logger/library/src/main/java/com/abcxyz/lumberjack/auditlogclient/AuditLoggingServerInterceptor.java [hunk 2]
```diff
@@ -289,4 +310,20 @@ Optional<Selector> getRelevantSelector(String methodIdentifier) {
...
+   * Get the justification token out of the headers, and create a struct of the correct format for
+   * use in audit logging metadata. We expect the claims from the JWT to be put in JSON format and
+   * put into the struct under the justification_token key.
+   */
+  Struct getStructForJustification(Metadata headers) throws JwkException {
+    Metadata.Key<String> metadataKey =
+        Metadata.Key.of("justification_token", Metadata.ASCII_STRING_MARSHALLER);
+    String jwtString = headers.get(metadataKey);
+    DecodedJWT jwt = jvs.validateJWT(jwtString);
+    String jsonString = StringUtils.newStringUtf8(Base64.decodeBase64(jwt.getPayload()));
```
Comment (sethvargo) [inline clients/java-logger/library/src/main/java/com/abcxyz/lumberjack/auditlogclient/AuditLoggingServerInterceptor.java:324]: I don't know a lot about Java, but does this handle all the possible b64 encoding schemes automatically?

Comment (raserva) [reply clients/java-logger/library/src/main/java/com/abcxyz/lumberjack/auditlogclient/AuditLoggingServerInterceptor.java:324]: Yes, it should decode to UTF8 all base64 encoded strings. https://commons.apache.org/proper/commons-codec/apidocs/org/apache/commons/codec/binary/Base64.html

**Comment by raserva** [comment]
The newest push build on the following PR: https://github.com/abcxyz/jvs/pull/92 and shouldn't be pushed until that one is.

File: clients/java-logger/library/src/main/java/com/abcxyz/lumberjack/auditlogclient/AuditLoggingServerInterceptor.java (hunk 2 above)
Comment (sethvargo) [reply clients/java-logger/library/src/main/java/com/abcxyz/lumberjack/auditlogclient/AuditLoggingServerInterceptor.java:324]: I see:

> The URL-safe parameter is only applied to encode operations. Decoding seamlessly handles both modes.

But I don't see anything about decoded padded vs unpadded. Maybe let's add a test with a few different base64 formats to be sure?

These should all decode to `foor4?-`:

```text
Zm9vcjQ/LQo=
Zm9vcjQ_LQo=
Zm9vcjQ/LQo
Zm9vcjQ_LQo
```

Comment (raserva) [reply clients/java-logger/library/src/main/java/com/abcxyz/lumberjack/auditlogclient/AuditLoggingServerInterceptor.java:324]: I put together a quick test.

```
 public static void main(String[] args) {
    List<String> strings = new ArrayList<>();
    String compareTo = StringUtils.newStringUtf8(Base64.decodeBase64("Zm9vcjQ/LQo="));
    strings.add(StringUtils.newStringUtf8(Base64.decodeBase64("Zm9vcjQ_LQo=")));
    strings.add(StringUtils.newStringUtf8(Base64.decodeBase64("Zm9vcjQ/LQo")));
    strings.add(StringUtils.newStringUtf8(Base64.decodeBase64("Zm9vcjQ_LQo")));

    for (String string : strings) {
      if (!compareTo.equals(string)) {
        throw new IllegalStateException("NOT EQUAL!");
      }
    }
  }
```

I can confirm the above code doesn't throw an exception, and if i print the strings, they all end up as the string `foor4?-`

**Review by sethvargo — APPROVED** [review]

**Pull request merged**